claripy/frontends/portfolio_frontend.py

```diff
@@ -0,0 +1,254 @@
+import logging
+from multiprocessing import Process, Pool
+from functools import partial
+
+from .constrained_frontend import ConstrainedFrontend
+from ..frontend import Frontend
+
+import traceback
+
+l = logging.getLogger("claripy.frontends.portfolio_frontend")
+
+
+def execute_solver_satisfiable(args):
+    solver, extra_constraints, exact = args
+    try:
+        satness = solver.check_satisfiability(
+            extra_constraints=extra_constraints,
+            exact=exact
+        )
+        if satness == 'UNKNOWN':
+            return solver, None
+        return solver, satness == 'SAT'
+    except Exception as e:
+        traceback.print_exc()
+        l.error("Solver failed to check satisfiability because %r", e)
+        return solver, None
+
+
+def execute_solver_eval(args):
+    solver, e, n, extra_constraints, exact = args
+    try:
+        res = solver.eval(
+            e, n, extra_constraints=extra_constraints, exact=exact
+        )
+        return solver, res
+    except Exception as e:
+        traceback.print_exc()
+        l.error("Solver failed to get solution because %r", e)
+        return solver, None
+
+
+class PortfolioFrontend(Frontend):
```

Comment:
Sounds good to me!

Reply:
Are we going to leave Portfolio solver out for this PR? I'm fine with merging it in as it is -- obviously if we don't have test cases, it will definitely break soon...